hedera-node/hedera-evm-api/src/main/java/com/hedera/services/evm/store/contracts/HederaEvmWorldState.java

```diff
@@ -0,0 +1,101 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.services.evm.store.contracts;
+
+import com.hedera.services.evm.accounts.AccountAccessor;
+import com.hedera.services.evm.contracts.execution.EvmProperties;
+import java.util.stream.Stream;
+import org.apache.tuweni.bytes.Bytes32;
+import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.datatypes.Hash;
+import org.hyperledger.besu.datatypes.Wei;
+import org.hyperledger.besu.evm.account.Account;
+
+public class HederaEvmWorldState implements HederaEvmMutableWorldState {
+
+    private final HederaEvmEntityAccess hederaEvmEntityAccess;
+    private final EvmProperties evmProperties;
+    private final AbstractCodeCache abstractCodeCache;
+
+    AccountAccessor accountAccessor;
+
+    protected HederaEvmWorldState(
+            HederaEvmEntityAccess hederaEvmEntityAccess,
+            EvmProperties evmProperties,
+            AbstractCodeCache abstractCodeCache) {
+        this.hederaEvmEntityAccess = hederaEvmEntityAccess;
+        this.evmProperties = evmProperties;
+        this.abstractCodeCache = abstractCodeCache;
+    }
+
+    protected HederaEvmWorldState(
+            HederaEvmEntityAccess hederaEvmEntityAccess,
+            EvmProperties evmProperties,
+            AbstractCodeCache abstractCodeCache,
+            AccountAccessor accountAccessor) {
+        this(hederaEvmEntityAccess, evmProperties, abstractCodeCache);
+        this.accountAccessor = accountAccessor;
+    }
+
+    public Account get(final Address address) {
+        if (address == null) {
+            return null;
+        }
+        if (hederaEvmEntityAccess.isTokenAccount(address)
+                && evmProperties.isRedirectTokenCallsEnabled()) {
+            return new HederaEvmWorldStateTokenAccount(address);
+        }
+        if (!hederaEvmEntityAccess.isUsable(address)) {
+            return null;
+        }
+        final long balance = hederaEvmEntityAccess.getBalance(address);
+        return new WorldStateAccount(
+                address, Wei.of(balance), abstractCodeCache, hederaEvmEntityAccess);
+    }
+
+    @Override
+    public Hash rootHash() {
+        return Hash.EMPTY;
+    }
+
+    @Override
+    public Hash frontierRootHash() {
+        return rootHash();
+    }
+
+    @Override
+    public Stream<StreamableAccount> streamAccounts(Bytes32 startKeyHash, int limit) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public HederaEvmWorldUpdater updater() {
+        return new Updater(accountAccessor);
+    }
+
+    public static class Updater extends AbstractLedgerEvmWorldUpdater
+            implements HederaEvmWorldUpdater {
+
+        protected Updater(AccountAccessor accountAccessor) {
+            super(accountAccessor);
```

Comment:
Here we should initialize a new object instead of triggering parent constructor. Ideally we would want to create new instance of `Stacked` updater e.g. a parent for `HederaStackedWorldStateUpdater`.